flink-streaming-connectors/flink-connector-elasticsearch2/src/main/java/org/apache/flink/streaming/connectors/elasticsearch2/helper/ElasticSearchHelper.java

```diff
@@ -0,0 +1,185 @@
+		for (TransportAddress transport : transportNodes) {
+			transportClient.addTransportAddress(transport);
+		}
+
+		// verify that we actually are connected to a cluster
+		ImmutableList<DiscoveryNode> nodes = ImmutableList.copyOf(transportClient.connectedNodes());
+		if (nodes.isEmpty()) {
+			throw new RuntimeException("Client is not connected to any Elasticsearch nodes!");
+		}
+		return transportClient;
+	}
+
+	/**
+	 * Create a new index template.
+	 * 
+	 * @param templateName Name of the template to create
+	 * @param templateReq Json defining the index template
+	 */
+	public void initTemplate(String templateName, String templateReq) throws Exception {
+		// Check if the template is set
+		if (templateReq != null && !templateReq.equals("")) {
+			// Json deserialization
+			PutIndexTemplateRequest request = new PutIndexTemplateRequest(templateName).source(templateReq);
+			// Sending the request to elastic search
+			sendIndexTemplateRequest(request);
+		}
+	}
+
+	/**
+	 * Send the index template request to elasticsearch.
+	 * 
+	 * @param indexTemplateRequest A valid index template request
+	 */
+	public void sendIndexTemplateRequest(PutIndexTemplateRequest indexTemplateRequest) throws Exception {
+		// Check if the template is set
+		if (indexTemplateRequest != null) {
+			// Sending the request to elastic search
+			client.admin().indices().putTemplate(indexTemplateRequest).get();
+		}
+	}
+
+	/**
+	 * Create a new mapping for a document type for an index.
+	 * 
+	 * @param indexName Index name where add the mapping
+	 * @param docType Document type of the mapping
+	 * @param mappingReq Json defining the index mapping
+	 */
+	public void initIndexMapping(String indexName, String docType, String mappingReq) throws Exception {
+		PutMappingRequest request=new PutMappingRequest(indexName).source(mappingReq).type(docType);
+		
+		// Put the mapping to the index
+		sendIndexMappingRequest(request);
+		LOG.debug("Updating mappings...");
+	}
+	
+	/**
+	 * Send the index mapping request to elasticsearch.
+	 * 
+	 * @param mappingRequest A valid index mapping request
+	 */
+	public void sendIndexMappingRequest(PutMappingRequest mappingRequest) throws Exception {
+		// Check if the template is set
+		if (mappingRequest != null) {
+			try {
+				// Check if the index exists
+				SearchResponse response = client.prepareSearch(mappingRequest.indices())
+						.setTypes(mappingRequest.type()).get();
+				if (response != null) {
+					LOG.debug("Index found, no need to create it...");
+				}
+			} catch (IndexNotFoundException infe) {
+				for (String indexName:mappingRequest.indices()){
+					// If the index does not exist, create it
+					client.admin().indices().prepareCreate(indexName)
+							.setSettings(Settings.builder().put("index.number_of_shards", DEFAULT_INDEX_SHARDS)
+							.put("index.number_of_replicas", DEFAULT_INDEX_REPLICAS)).execute().actionGet();
```

Comment:
Use elasticsearch constant for "index.number_of_replicas" => IndexMetaData. SETTING_NUMBER_OF_REPLICAS